### PR TITLE
Form access control votes

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Discussed lengthy internally with @cedric-anne, replacement for https://github.com/glpi-project/glpi/pull/17117.

TLDR: We redefine the default behavior for our forms access controls policies while allowing future plugins to do things differently. No interface changes and no new configuration options.

----

As a reminder we have two possible policies:
* Allow specific users/groups/profiles
* Allow direct token access

Before this PR, enabling both these criteria would lead to an AND condition (user must have valid token AND be whitelisted) which is not intuitive for users because the names specify "Allow xxx" and not "Restrict xxx".
Thus, this has been changed to an OR condition.

Internally, we use a voting system with 3 choices (grant, abstain, deny).
Our default policies only use the grant/abstain vote but future plugins may want to add their own access policies and use the deny vote (for example, a "Restrict by IP" policy).

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: src/Glpi/Form/AccessControl/AccessVote.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * vote couples:
+ * - "Allow access by specific IP": (Grant, Abstain)
+ * - "Restrict access by specific IP": (Grant, Deny)
+ * - "Block access to specific IP": (Abstain, Deny)
+ */
+enum AccessVote
+{
+    case Grant;
+    case Abstain;
+    case Deny;
```
Comment (cedric-anne) [inline src/Glpi/Form/AccessControl/AccessVote.php:56]: Some comments could help developpers.

```suggestion
    /**
      * Grant access.
      * The access grant can still be denied if any other policy vote is `Deny`.
      */
    case Grant;

    /**
      * Does not allow access, but does not deny it either.
      * Access grant will depend on other policies vote.
      */
    case Abstain;

    /**
      * Deny access.
      * The access grant will be refused, whatever the other policies vote is.
      */
    case Deny;
```

**Review by cedric-anne — APPROVED** [review]
